pgjdbc/src/main/java/org/postgresql/util/internal/IntSet.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2024, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.util.internal;
+
+import org.postgresql.core.Oid;
+
+import java.util.BitSet;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Read-optimized {@code Set} for storing {@link Oid} values.
+ */
+public class IntSet {
+  /**
+   * Maximal Oid that will bs stored in {@link BitSet}.
+   * If Oid exceeds this value, then it will be stored in {@code Set<Int>} only.
+   * In theory, Oids can be up to 32bit, so we want to limit per-connection memory utilization.
+   * Allow {@code BitSet} to consume up to 8KiB (one for send and one for receive).
+   */
+  private static final int MAX_OID_TO_STORE_IN_BITSET = 8192 * 8;
+
+  /**
+   * Contains all the values.
+   * Note: we could skip storing small values in ths set, however,
```

Comment:
How often is asSet() actually called?
How often do expect to provide binary support for oids beyond 64k?